### ISSUE TITLE
Keys in FileKache are erroneously deleted

### ISSUE THREAD

**Issue by vfsfitvnm** [MayakaApps/Kache#173]
**Describe the bug**
It looks like the cache is getting trimmed without a reason:

**To Reproduce**
Consider this test:
```kt
    @Test
    fun issue173() = runTest {
        val fileSystem = FakeFileSystem()

        val kache = testOkioFileKache(fileSystem)

        for (i in 0 until 16) {
            kache.put(i.toString()) { cachePath ->
                fileSystem.sink(cachePath).buffer().use { it.writeByte(0) }
                true
            }
        }

        assertEquals(16, kache.size)
        assertEquals(16, kache.getKeys().size)
    }
```
It writes 16 keys for a total of 16 bytes (the cache max size is 1024).

**However, the last test fails**: but there should be 16 keys!

**Comment by vfsfitvnm** [comment]
Whoops, I clicked the submit button a little early - let me edit the issue...

**Comment by MSDarwish2000** [comment]
Thank you for reporting this serious problem. It also affects `InMemoryKache`. I'll investigate it and release a fix ASAP.

**Comment by MSDarwish2000** [comment]
@vfsfitvnm The issue was resolved in v2.1.0-beta03. Thank you for reporting it.
